plugin/src/test/java/org/opensearch/ml/rest/RestMLInferenceIngestProcessorIT.java

```diff
@@ -350,6 +361,108 @@ public void testMLInferenceProcessorWithForEachProcessor() throws Exception {
+            + "        \"type\": \"nested\",\n"
+            + "        \"properties\": {\n"
+            + "          \"title_embedding\": {\n"
+            + "            \"type\": \"float\"\n"
+            + "          },\n"
+            + "          \"title\": {\n"
+            + "            \"type\": \"text\"\n"
+            + "          },\n"
+            + "          \"description\": {\n"
+            + "            \"type\": \"text\"\n"
+            + "          }\n"
+            + "        }\n"
+            + "      }\n"
+            + "    }\n"
+            + "  }\n"
+            + "}";
+        createIndex(indexName, createIndexRequestBody);
+
+        String createPipelineRequestBody = "{\n"
+            + "  \"description\": \"test embeddings\",\n"
+            + "  \"processors\": [\n"
+            + "    {\n"
+            + "      \"foreach\": {\n"
+            + "        \"field\": \"books\",\n"
+            + "        \"processor\": {\n"
+            + "          \"ml_inference\": {\n"
+            + "            \"model_id\": \""
+            + localModelId
+            + "\",\n"
+            + "            \"input_map\": [\n"
+            + "              {\n"
+            + "                \"input\": \"_ingest._value.title\"\n"
+            + "              }\n"
+            + "            ],\n"
+            + "            \"output_map\": [\n"
+            + "              {\n"
+            + "                \"_ingest._value.title_embedding\": \"$.embedding\"\n"
+            + "              }\n"
+            + "            ],\n"
+            + "            \"ignore_missing\": false,\n"
+            + "            \"ignore_failure\": false\n"
+            + "          }\n"
+            + "        }\n"
+            + "      }\n"
+            + "    }\n"
+            + "  ]\n"
+            + "}";
+        createPipelineProcessor(createPipelineRequestBody, pipelineName);
+
+        String uploadDocumentRequestBody = "{\n"
+            + "    \"books\": [{\n"
+            + "            \"title\": \"first book\",\n"
+            + "            \"description\": \"This is first book\"\n"
+            + "        },\n"
+            + "        {\n"
+            + "            \"title\": \"second book\",\n"
+            + "            \"description\": \"This is second book\"\n"
+            + "        }\n"
+            + "    ]\n"
+            + "}";
+        uploadDocument(indexName, "1", uploadDocumentRequestBody);
+        Map document = getDocument(indexName, "1");
+
+        List embeddingList = JsonPath.parse(document).read("_source.books[*].title_embedding");
+        Assert.assertEquals(2, embeddingList.size());
+
+        List embedding1 = JsonPath.parse(document).read("_source.books[0].title_embedding");
+        Assert.assertEquals(1536, embedding1.size());
+        List embedding2 = JsonPath.parse(document).read("_source.books[1].title_embedding");
+        Assert.assertEquals(1536, embedding2.size());
```

Comment:
for the IT, this test for each processor with nested documents, can you also add the test that do not use for each processor? does it still work fine?

Reply:
Yes it works fine. Will add few more tests for local models